packages/techdocs-common/src/stages/generate/helpers.ts

```diff
@@ -138,6 +139,45 @@ export const getRepoUrlFromLocationAnnotation = (
   return undefined;
 };
 
+/**
+ * Validating mkdocs config file for incorrect/insecure values
+ * Return a boolean value defining if mkdocs.yml within the location is valid
+ *
+ * @param {string} mkdocsYmlPath Absolute path to mkdocs.yml or equivalent of a docs site
+ * @param {Logger} logger
+ */
+export const validateMkdocsYaml = async (
+  mkdocsYmlPath: string,
+  logger: Logger,
+): Promise<boolean> => {
+  let mkdocsYmlFileString;
+  try {
+    mkdocsYmlFileString = await fs.readFile(mkdocsYmlPath, 'utf8');
+  } catch (error) {
+    logger.error(
+      `Could not read MkDocs YAML config file ${mkdocsYmlPath} before for validation: ${error.message}`,
+    );
+    return false;
+  }
+
+  let mkdocsYml: any;
+  try {
+    mkdocsYml = yaml.load(mkdocsYmlFileString);
+
+    if (mkdocsYml.docs_dir && isAbsolute(normalize(mkdocsYml.docs_dir))) {
+      throw new Error(
+        "docs_dir configuration value in mkdocs can't be an absolute directory path for security reasons. Use relative paths instead which are resolved relative to your mkdocs.yml file location.",
+      );
+    }
+  } catch (error) {
+    logger.error(
```

Comment:
Let's just straight up throw an error and let it propagate out of this function imo, i.e. make the return type `Promise<void>`. The throw, catch, log, return, throw is a bit awkward and hides information. I don't think there's any reason to hide the full error from the user right?

Reply:
Yup, makes sense :+1: tweaked